aexpect/ops.py

```diff
@@ -0,0 +1,601 @@
+and system utilities.
+
+stat:
+Allows passing an optional format argument (``-c``). Custom wrappers for
+extracting *atime*, *ctime*, and *mtime*, etc exist.
+
+test:
+Allows for testing existence, permissions and other properties of files.
+
+file ops:
+Simple file operations executed through sessions. Most functions boil down to
+executing simple commands like ls, tar, md5sum, etc. If these operations return
+non-0, a :py:class:`RuntimeError` is raised, containing the commands's error
+message. All functions :py:func:`shlex.quote` their args for better security.
+
+utilities:
+More complex linux utilities for tarball extraction, file hashing, etc.
+
+system:
+System related operations like restarting or stopping services.
+"""
+
+import logging
+import re
+from os.path import join
+from os.path import split as path_split
+from shlex import quote
+
+from aexpect.exceptions import ShellCmdError
+
+# Need this import for sphinx and other documentation to produce links later on
+# from .client import ShellSession
+
+
+###############################################################################
+# stat(2)
+###############################################################################
+
+
+def stat(session, path, fmt=None):
+    """
+    Wrapper for ``stat``.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str path: path to the filesystem entry to stat
+    :param str fmt: optional format passed to ``-d``
+    :returns: standard output of stat(1) on ``path``
+    :rtype: str
+    """
+    if fmt is None:
+        return session.cmd_output(r"stat '%s'" % path)
+
+    if not isinstance(fmt, str) or fmt[0] != "%":
+        raise RuntimeError("%s is not a valid format string for stat(1)" % fmt)
+
+    return session.cmd_output(r"stat -c %s '%s'" % (fmt, path))
+
+
+def get_atime(session, path):
+    """
+    Query the access time.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str path: path to the filesystem entry to stat
+    :returns: standard output of stat(1) querying the access time of ``path``
+    :rtype: str
+    """
+    return stat(session, path, fmt="%X")
+
+
+def get_mtime(session, path):
+    """
+    Query the modification time.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str path: path to the filesystem entry to stat
+    :returns: standard output of stat(1) querying the last modification time
+              of ``path``
+    :rtype: str
+    """
+    return stat(session, path, fmt="%z")
+
+
+def get_ctime(session, path):
+    """
+    Query the change time.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str path: path to the filesystem entry to stat
+    :returns: standard output of stat(1) querying the last change time of
+              ``path``
+    :rtype: str
+    """
+    return stat(session, path, fmt="%y")
```

Comment:
And here you also return human-readable format, different from `get_atime`, but like `get_mtime`.

Reply:
Alright, fixing this in the next push, will resolve three of your comments afterwards so feel free to comment again if it doesn't fix what you meant.